docs/WhatAndWhy/enhancedmemoryprotection.md

```diff
@@ -0,0 +1,159 @@
+
+Many bootloaders and OPROMs will not have implemented support for enhanced protections on
+image memory, allocated buffers, and other memory ranges. To indicate support for enhanced
+protections, the PE/COFF IMAGE_DLLCHARACTERISTICS_NX_COMPAT DLL characteristic will be used.
+Modules with this characteristic are expected to be compliant with enhanced memory protection
+and should utilize the Memory Attribute Protocol to manipulate the attributes of memory they
+allocate. If a module is loaded without this characteristic, the platform should enter
+compatibility mode.
+
+## Compatibility Mode
+
+To provide a more consistent and predictable environment across UEFI implementations,
+we are providing a definition for compatibility mode here. Errant modules and unexpected
+faults blocking boot should enter compatibility mode which triggers the following
+deviations from the enhanced memory protection definition:
+
+1. All new memory allocated will be readable, writable, and executable.  
+2. All images loaded from the start of compatibility mode will no longer have
+restrictive access attributes applied to the memory ranges in which they are loaded.  
+3. The Memory Attribute Protocol will be uninstalled.  
+4. Page zero will be mapped.  
+5. Legacy BIOS memory (the lower 640K range) will be mapped as readable, writable, and
+executable.
+
+### A Note on User Notification of Compatibility Mode
+
+When a system is in compatibility mode, it is important that the user receives some
+notification during boot which could take the form of a dialogue box, color bar, or
+other visual indicator. At the start of this journey toward more strict memory
+protections, we believe how compatibility mode is visualized should be left up to the
+platform. In the future, platform developers should consider adding a TPM (Trusted
+Platform Module) measurement when the system enters compatibility mode to have platform
+enforcement of memory protections.
+
+## Closing
+
+It will take substantial work to update legacy code to adhere to these new security
+standards. These protection mitigations offer significant value to end users by
+heightening both real and perceived security, and to OEMs by reducing the number
+of issues in the future. As we push toward updating incompatible modules, OEMs
+should apply strict protections during product development which meet or exceed the
+standard outlined in this document to help catch programmer error and reduce the
+volume of unsafe firmware code entering the ecosystem. Microsoft is committed to
+this effort and is prepared to work with partners as we move toward a more
+secure UEFI ecosystem.
+
+## Additional Resources
+
+PE Format: PE Format - Win32 apps | Microsoft Learn
+Requirements for 3rd party signing: UEFI CA Memory Mitigation Requirements for Signing - Windows drivers | Microsoft Learn
+UEFI Test Tool: mu_plus/UefiTestingPkg/AuditTests/PagingAudit at release/202311 · microsoft/mu_plus (github.com)

```

Comment:
A few notes about these specific links, `en-us` should be dropped from learn.microsoft.com links so they can localize for a given user. `HEAD` is used as the ref instead of a branch name in git repos to keep current.

Reply:
Thanks for fixing the links :)